From a742042cb7273fbe786eb5af7a4052214b29a136 Mon Sep 17 00:00:00 2001
From: Hatem ElKharashy <hatem.elkharashy@qt.io>
Date: Mon, 27 Oct 2025 10:39:19 +0200
Subject: Fix gradient stops current color parsing

After the refactor done to CSS parsing, current color for gradient
stops was no longer checked and led to missing style of gradient
stops.

Fixes: QTBUG-141102
Change-Id: I80385164b0c2eebbf18abe887468105d145f317f
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
(cherry picked from commit a84a367a8d779b8aa4ba811ede1dff310d5091f7)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/svg/qsvghandler.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/svg/qsvghandler.cpp b/src/svg/qsvghandler.cpp
index 0066a89..14faed6 100644
--- a/src/svg/qsvghandler.cpp
+++ b/src/svg/qsvghandler.cpp
@@ -3420,6 +3420,9 @@ static bool parseStopNode(QSvgStyleProperty *parent,
     attrs.setAttributes(styleCssAttributes, handler);
 #endif
 
+    //TODO: Handle style parsing for gradients stop like the rest of the nodes.
+    parseColor(&dummy, attrs, handler);
+
     QSvgGradientStyle *gradientStyle =
         static_cast<QSvgGradientStyle*>(parent);
     QStringView colorStr    = attrs.stopColor;
-- 
cgit v1.2.3

